### ISSUE TITLE
Creating stubs visible by reflection with MockK

### ISSUE THREAD

**Issue by oleksiyp** [mockk/mockk#149]
https://stackoverflow.com/questions/52497324/creating-stubs-visible-by-reflection-with-mockk

**Comment by Kerooker** [comment]
I don't believe this should be added.

It's a way too specific requirement, and a bad testing pattern anyways. I don't believe it should be encouraged

**Comment by oleksiyp** [comment]
First thing I do not understand all the requirements. All the judging afterwards

**Comment by VitekKlugi** [comment]
Thank you, for creating this issue. How can I help to clarify the requirements? Which specific requirements are not clear?

**Comment by oleksiyp** [comment]
@VitekKlugi so this framework is using fields instead of properties to fetch data for comparison? If it would compare it via getters there most probable won't be any problems

**Comment by VitekKlugi** [comment]
@oleksiyp yes, you are right the framework iterates through the fields of the objects recursively using reflection and compares them for equality. We will try to look for different framework in our tem that would use getters instead. Thanks anyway for you effort.

**Comment by oleksiyp** [comment]
Please verify the fix in version 1.8.11 or 1.8.11.kotlin13.

I added the possibility to set underlying field value by simple construct `every { a.property } returns value`. This may be useful in some other cases as well as yours. Do not know if library going to work in this case

**Comment by VitekKlugi** [comment]
Thank you for the fix. It works great for the mocked Kotlin classes. But it seems to me that for some reason it doesn't work for mocked Java classes. I'll try to do more research later.

**Comment by VitekKlugi** [comment]
New observation: if the Java class is final it seems to work, otherwise I get java.lang.NoSuchFieldException.
